baselines/fedexp/fedexp/models.py

```diff
@@ -0,0 +1,222 @@
+            self.layers.extend(block_layers)
+        self.linear = nn.Linear(512 * block.expansion, num_classes)
+
+    def forward(self, x):
+        """Forward pass through the network."""
+        out = F.relu(self.bn1(self.conv1(x)))
+        for layer in self.layers:
+            out = layer(out)
+        out = F.avg_pool2d(out, 3)
+        out = out.view(out.size(0), -1)
+        out = self.linear(out)
+        return out
+
+
+class ResNet18(_ResNet):
+    """ResNet18 model."""
+
+    def __init__(self, num_classes=10):
+        super().__init__(num_classes=num_classes, num_blocks=[2, 2, 2, 2])
+
+
+def test(
+        net: nn.Module, test_loader: DataLoader, device: torch.device
+) -> Tuple[float, float]:
+    """Evaluate the network on the entire test set.
+
+    Parameters
+    ----------
+    net : nn.Module
+        The neural network to test.
+    test_loader : DataLoader
+        The DataLoader containing the data to test the network on.
+    device : torch.device
+        The device on which the model should be tested, either 'cpu' or 'cuda'.
+
+    Returns
+    -------
+    Tuple[float, float]
+        The loss and the accuracy of the input model on the given data.
+    """
+    if len(test_loader.dataset) == 0:
+        raise ValueError("Testloader can't be 0, exiting...")
+
+    criterion = torch.nn.CrossEntropyLoss()
+    correct, total, loss = 0, 0, 0.0
+    net.eval()
+    with torch.no_grad():
+        for images, labels in tqdm(test_loader, "Testing ..."):
+            images, labels = images.to(device), labels.to(device)
+            outputs = net(images)
+            loss += criterion(outputs, labels).item()
+            _, predicted = torch.max(outputs.data, 1)
+            total += labels.size(0)
+            correct += (predicted == labels).sum().item()
+    loss /= len(test_loader.dataset)
+    accuracy = correct / total
+    return loss, accuracy
+
+
+def train(  # pylint: disable=too-many-arguments
+        net: nn.Module,
+        trainloader: DataLoader,
+        device: torch.device,
+        epochs: int,
+        hyperparams: Dict[str, Scalar],
+) -> None:
+    """Train the network on the training set.
+
+    Parameters
+    ----------
+    net : nn.Module
+        The neural network to train.
+    trainloader : DataLoader
+        The DataLoader containing the data to train the network on.
+    device : torch.device
+        The device on which the model should be trained, either 'cpu' or 'cuda'.
+    epochs : int
+        The number of epochs the model should be trained for.
+    hyperparams : Dict[str, Scalar]
+        The hyperparameters to use for training.
+    """
+    criterion = torch.nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(
+        net.parameters(),
+        lr=hyperparams["eta_l"],
+        momentum=0,
+        weight_decay=hyperparams["weight_decay"],
+    )
+    net.train()
+    for _ in tqdm(range(epochs), desc="Local Training ..."):
```

Comment:
I think it's better to remove `tqdm` from here since if multiple clients run in parallel it will lead to not-very-useful logs.... It's great to have `tqdm` for the server evaluation though.